queue_job/jobrunner/runner.py

```diff
@@ -343,6 +327,48 @@ def set_job_enqueued(self, uuid):
                 (ENQUEUED, uuid),
             )
 
+    def requeue_dead_jobs(self):
+        """
+        Set started and enqueued jobs but not locked to pending
+
+        A job is locked when it's being executed
+        When a job is killed, it releases the lock
+
+        Adding a buffer on 'date_enqueued' to check
+        that it has been enqueued for more than 10sec.
+        This prevents from requeuing jobs before they are actually started.
+
+        When Odoo shuts down normally, it waits for running jobs to finish.
+        However, when the Odoo server crashes or is otherwise force-stopped,
+        running jobs are interrupted while the runner has no chance to know
+        they have been aborted.
+        """
+        query = """
+            UPDATE
+                queue_job
+            SET
+                state='pending',
+                retry=retry+1
+            WHERE
+                id IN (
+                    SELECT
+                        job.id
+                    FROM
+                        queue_job_locks AS lock
+                        LEFT JOIN
+                            queue_job AS job
+                        ON
+                            lock.id = job.id
+                    WHERE
+                        job.state  in ('started','enqueued')
+                        AND date_enqueued <
+                        date_trunc('seconds',now() AT TIME ZONE 'utc' - INTERVAL '10 sec')
```

Comment:
Is date_trunc necessary?

Reply:
`date_enqueued` is truncated to the seconds but yes, it is not necessary.
I don't see it as an issue, do you want me to remove it?

Reply:
If it's not needed let's remove it. It will be more performant if anything.